### DISCUSSION TITLE
Should we drop Affect?

### DISCUSSION THREAD

**Discussion by baetheus** [baetheus/fun#62] [Ideas]
I've built a few cli tools in the past couple of months using fun and whenever I use Affect I find myself bound by the simplified format in interesting ways. I had intended for Affect to be a nicer bridge between non-fp deno libs and fp style tools, but I found that for [some patterns](https://github.com/nullpub/fun/blob/flake/scripts/build-npm.ts) it is much nicer to use TaskEither. In fact, I've yet to find a case where I'd rather use Affect over TaskEither.

Another issue is that purescript has Aff (which I realized after making the Affect adt) and it implements cancellation. While Deno does have abort controller, it would make sense for Affect to implement cancellation. Since it doesn't the interface feels lacking to me.

@baoshan @pixeleet What do you guys think? Have you used Affect? Do you think we should keep it?

**Comment by baoshan** [comment]
I have not used Affect yet. I use Task and TaskEither as a bridge to cross the boundaries I met.

—— Sent from my baby walker

**Comment by pixeleet** [comment]
TaskEither, ReaderTaskEither and StateReaderTaskEither for the win! 
Definitely in favor of moving closer to fp-ts implementation for easier porting.

**Reply by baetheus** [reply]
This sounds great. I'll slate those for after 2.0

**Reply by pixeleet** [reply]
Missed this comment in the flurry of incoming notifications. I see you're pretty busy with hashing 2.0 out. 
Was just about to open a discussionn about getting RTE/SRTE in, than I remembered we talked about it. 

Would be keen to get a crack on RTE/SRTE

**Reply by baetheus** [reply]
Sorry about the spam : /. I figured that since a single big documentation ticket wasn't working that I should break things into clearer chunks.

Anyway, I'm totally up for you taking a crack at ReaderTaskEither and StateReaderTaskEither. I've never really used that Big of an ADT for anything in javascript, but I could see it being useful for some folks. I'd say start by reading through fp-ts [ReaderTaskEither](https://github.com/gcanti/fp-ts/blob/master/src/ReaderTaskEither.ts) but there is a lot of noise there. Instead, I would recommend looking at implementing all of the combinators, instances, and instance getters in our [ReaderEither](https://github.com/baetheus/fun/blob/main/reader_either.ts) for ReaderTaskEither. I'll probably implement a few helper functions for type classes like [this](https://gist.github.com/baetheus/20b0c3e8ca1ad0c40573278d3851315f) as I burn through this 2.0 march.

As an aside, you're welcome to pick up any of the tickets in the queue that interest you. You never have to ask for permission and you can email me if you hit any snags for help.

**Comment by baetheus** [comment]
I've dropped affect.ts in main.
